src/genericComponents/components.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useRef } from "react";
```

Comment:
Questo componente lo rinominerei in Menu tirerei fuori upperHeader e creerei un nuovo componente Header dove ingloberei sia l'upperHeader e sia il componente Menu se è situato nel' header ovviamente (ora non ricordo :) ), ad ogni modo rinominerei components.js visto che è troppo generico.